core/store/src/lib.rs

```diff
@@ -296,7 +300,7 @@ impl Store {
     }
 
     pub fn store_update(&self) -> StoreUpdate {
-        StoreUpdate::new(Arc::clone(&self.storage))
+        StoreUpdate::new(self.storage.db_for_updates().unwrap_or(Arc::clone(&self.storage)))
```

Comment:
Nice that you only need to make the change here. Can you do a quick check to see if there aren't any other usages of storage for writing that don't use store_update? Ideally this would be enforced by rust type system checking mutability. Implementing it in this PR would be a stretch but if this happens to be the case already that's great.